lib/big_brother/cluster.rb

```diff
@@ -1,6 +1,15 @@
 module BigBrother
   class Cluster
-    attr_reader :fwmark, :scheduler, :check_interval, :nodes, :name, :ramp_up_time, :nagios
+    attr_reader :fwmark, :scheduler, :check_interval, :nodes, :name, :ramp_up_time, :nagios, :backend_mode
+    ACTIVE_PASSIVE_CLUSTER = "active_passive"
+
+    def self.create_cluster(name, attributes)
```

Comment:
@lollipopman A factory method provides a cleaner abstraction over messing around with the initializer in Cluster(which I am not sure how to return a different class/subclass from the initializer of the parent class) except I am misunderstanding you.